test/simulator/main/org/apache/cassandra/simulator/ClusterSimulation.java

```diff
@@ -141,6 +141,21 @@ public interface SchedulerFactory
         RunnableActionScheduler create(RandomSource random);
     }
 
+    public interface FutureActionSchedulerFactory
+    {
+        FutureActionScheduler create(int nodeCount, SimulatedTime time, RandomSource random);
+    }
+
+    public interface PerVerbFutureActionSchedulersFactory
+    {
+        Map<Verb, FutureActionScheduler> create(int nodeCount, SimulatedTime time, RandomSource random);
+    }
+
+    public interface TimeFactory
+    {
+        SimulatedTime create(RandomSource random, int numOfNodes);
+    }
+
```

Comment:
going to revert this, this was an idea benedict and I played around with to try to repo the underline issue, but this was not actually needed to repo.

Reply:
removed